api/lib/infrastructure/repositories/organization-invitation-repository.js

```diff
@@ -1,58 +1,39 @@
 import { knex } from '../../../db/knex-database-connection.js';
 import _ from 'lodash';
 
-function _toDomain(bookshelfInvitation) {
-  if (bookshelfInvitation) {
-    return bookshelfToDomainConverter.buildDomainObject(BookshelfOrganizationInvitation, bookshelfInvitation);
-  }
-  return null;
-}
-
-function _checkNotFoundError(error, id) {
-  if (error instanceof BookshelfOrganizationInvitation.NotFoundError) {
-    throw new NotFoundError(`Not found organization-invitation for ID ${id}`);
-  }
-  throw error;
-}
-
-function _checkNotFoundErrorWithCode({ error, id, code }) {
-  if (error instanceof BookshelfOrganizationInvitation.NotFoundError) {
-    throw new NotFoundError(`Not found organization-invitation for ID ${id} and code ${code}`);
-  }
-  throw error;
-}
-
-const create = function ({ organizationId, email, code, role }) {
+const create = async function ({ organizationId, email, code, role }) {
   const status = OrganizationInvitation.StatusType.PENDING;
-  return new BookshelfOrganizationInvitation({ organizationId, email, status, code, role }).save().then(_toDomain);
+  const [organizationInvitationCreated] = await knex('organization-invitations')
+    .insert({ organizationId, email, status, code, role })
+    .returning('*');
+  if (!organizationInvitationCreated) return null;
+  return new OrganizationInvitation(organizationInvitationCreated);
 };
 
-const get = function (id) {
-  return BookshelfOrganizationInvitation.where({ id })
-    .fetch()
-    .then(_toDomain)
-    .catch((err) => _checkNotFoundError(err, id));
+const get = async function (id) {
+  const organizationInvitation = await knex('organization-invitations').where('id', id).first();
+  if (!organizationInvitation) throw new NotFoundError(`Not found organization-invitation for ID ${id}`);
+  return new OrganizationInvitation(organizationInvitation);
 };
 
-const getByIdAndCode = function ({ id, code }) {
-  return BookshelfOrganizationInvitation.where({ id, code })
-    .fetch()
-    .then(_toDomain)
-    .catch((error) => _checkNotFoundErrorWithCode({ error, id, code }));
+const getByIdAndCode = async function ({ id, code }) {
+  const organizationInvitation = await knex('organization-invitations').where({ id, code }).first();
+  if (!organizationInvitation)
+    throw new NotFoundError(`Not found organization-invitation for ID ${id} and code ${code}`);
+  return new OrganizationInvitation(organizationInvitation);
 };
 
-const markAsAccepted = function (id) {
+const markAsAccepted = async function (id) {
   const status = OrganizationInvitation.StatusType.ACCEPTED;
 
-  return new BookshelfOrganizationInvitation({ id })
-    .save({ status }, { patch: true, require: true })
-    .then((model) => model.refresh())
-    .then(_toDomain)
-    .catch((err) => _checkNotFoundError(err, id));
+  const [organizationInvitationAccepted] = await knex('organization-invitations')
+    .where({ id })
+    .update({ status })
+    .returning('*');
```

Comment:
Bookshelf pouvait se charger de modifier automatiquement le `updatedAt` d'un enregistrement, mais avec Knex c'est à faire manuellement à chaque requête effectuant une modification :

```suggestion
  const [organizationInvitationAccepted] = await knex('organization-invitations')
    .where({ id })
    .update({ status, updatedAt: new Date() })
    .returning('*');
```

Reply:
On va la rajouter, mais il serait intéressant que des tests soient rajoutés pour valider le `updatedAt` (peut être dans votre doc dette technique ? )

Reply:
Merci @xav-car ! Et bien reçu pour l'ajout de tests ! Je viens de créer un ticket TECH `PIX-10677` pour l'ajout de ces tests.